contracts/utils/contracts/src/LibEIP712.sol

```diff
@@ -46,21 +47,44 @@ contract LibEIP712 {
         pure
         returns (bytes32 result)
     {
-        return keccak256(abi.encodePacked(
-            EIP712_DOMAIN_SEPARATOR_SCHEMA_HASH,
-            keccak256(bytes(name)),
-            keccak256(bytes(version)),
-            chainId,
-            uint256(verifyingContractAddress)
-        ));
+        bytes32 schemaHash = _EIP712_DOMAIN_SEPARATOR_SCHEMA_HASH;
+
+        // Assembly for more efficient computing:
+        // keccak256(abi.encodePacked(
+        //     _EIP712_DOMAIN_SEPARATOR_SCHEMA_HASH,
+        //     keccak256(bytes(name)),
+        //     keccak256(bytes(version)),
+        //     chainId,
+        //     uint256(verifyingContractAddress)
+        // ))
+
+        assembly {
```

Comment:
This looks good to me ~ do you know roughly how much gas this saves?

Reply:
I don't think it's super significant, maybe 1000 or so. Also note that this function only gets called once in the `Exchange` contract. I mostly did this for consistency pre-vuln since all of the other hashing is done in assembly, but it would probably be a good time to formalize what we want to do with all of the hashing as a whole.